### PR TITLE
Added a new chapter/assembly about using foreman webhooks

### PR DESCRIPTION
**Pull request by suyogsainkar**
https://issues.redhat.com/browse/SAT-3106
[doc] Foreman Webhooks

Cherry-pick into:

* [ ] Foreman 2.5 (Satellite 6.10)
* [ ] Foreman 2.4
* [ ] Foreman 2.3 (Satellite 6.9)
* [ ] Foreman 2.1 (Satellite 6.8)

### CODE REVIEW

**Comment by suyogsainkar** [comment]
Hi @ianf77 when you get a chance could you take a look at this PR and make adjustments as needed? I know @melcorr would be happy to help if you have questions. Thanks!

File: guides/common/assembly_using-foreman-webhooks.adoc [hunk 1]
```diff
@@ -0,0 +1,27 @@
+ifdef::context[:parent-context: {context}]
+
+[id="using-foreman-webhooks_{context}"]
+= Using Foreman Webhooks 
+
+A webhook in web development is a method of augmenting or altering the behavior of a web page or web application with custom callbacks. Webhooks are "user-defined HTTP callbacks". They are usually triggered by some event, such as pushing code to a repository or a comment being posted to a blog. When that event occurs, the source site makes an HTTP request to the URL configured for the webhook. Users can configure them to cause events on one site to invoke behavior on another. Because webhooks use HTTP, they can be integrated into web services without adding new infrastructure.
```
Comment (lzap) [inline guides/common/assembly_using-foreman-webhooks.adoc:6]: This is taken straight from Wikipedia: https://en.wikipedia.org/wiki/Webhook which is fine but we need to give attribution somewhere. @melcorr I guess there is already a known process/place to put it?

File: guides/common/modules/proc_creating-a-webhook.adoc [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+. Select *Verify SSL* if the server certificate should be verified against the system certificate store or Foreman CA. When using shellhooks, ensure this flag is selected.
+. Select *Proxy Authorization* when using shellhooks, otherwise deselect this box.
+. Optional: Provide one or more CA certificates in the X509 Certification Authorities. Keep the field blank when using shellhooks.
+
+In the Additional tab:
+
+. Enter HTTP Content Type, typically application/json, application/xml or text/plain depending on the payload.
+. Optional: Provide HTTP headers as JSON. ERB is allowed and the same macros and variables are available as in webhook templates.
+
+When configuring webhooks with endpoints with non-standard HTTP(s) ports, SELinux port must be assigned, see <>.
```
Comment (lzap) [inline guides/common/modules/proc_creating-a-webhook.adoc:29]: Missing reference.

File: guides/common/modules/proc_installing-webhooks-shellhooks-plugin.adoc [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+ifdef::satellite[]
+
+* Install the webhooks plugin using the following command:
++
+[options="nowrap" subs="+quotes,attributes"]
+----
+# foreman-installer --enable-foreman-plugin-webhooks 
+----
+
+* Install the shellhooks plugin on each {SmartProxy} that will be used for shellhooks, using the following command:
```
Comment (lzap) [inline guides/common/modules/proc_installing-webhooks-shellhooks-plugin.adoc:15]: Optionally, install the shellhooks...

File: guides/common/modules/proc_installing-webhooks-shellhooks-plugin.adoc [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
++
+[options="nowrap" subs="+quotes,attributes"]
+----
+# foreman-installer --enable-foreman-proxy-plugin-shellhooks
+----
+
+* Install the CLI plugin using the following command:
```
Comment (lzap) [inline guides/common/modules/proc_installing-webhooks-shellhooks-plugin.adoc:22]: This is also optional.

File: guides/common/modules/proc_installing-webhooks-shellhooks-plugin.adoc [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+. Restart the {SmartProxy}.
++
+[options="nowrap" subs="+quotes,attributes"]
+----
+# systemctl restart foreman-proxy
+----
+
+In the {Project} web UI, navigate to *Infrastructure* > *Smart Proxies* > *Refresh features* to allow shellhooks feature to show up.
+
+endif::[]
```
Comment (lzap) [inline guides/common/modules/proc_installing-webhooks-shellhooks-plugin.adoc:59]: This foreman part is no longer valid, Foreman 2.4 is long gone. We can just use the installer part and remove "if satellite" from there.

**Comment by lzap** [comment]
From technical side, I am good.

**Comment by melcorr** [comment]
@lzap he no longer works with us as of Friday and didn't finish, so I'm hoping that @ianf77 can prioritise this. Otherwise we will merge as is  :-)

File: guides/common/assembly_using-foreman-webhooks.adoc (hunk 1 above)
Comment (ianf77) [reply guides/common/assembly_using-foreman-webhooks.adoc:6]: I would rather not borrow so obviously from Wikipedia. I'll  take a look at this initial paragraph. It's a bit wordy.

Comment (melcorr) [reply guides/common/assembly_using-foreman-webhooks.adoc:6]: Users need an introduction to the concept, why and how it is useful. @lzap added this to the google doc to give the writer a foundation. Suyog just copy pasted it rather than developing information.

Comment (ekohl) [reply guides/common/assembly_using-foreman-webhooks.adoc:6]: https://github.com/theforeman/foreman-documentation/pull/599 maybe that's a good place to start the discussion?

Comment (ianf77) [reply guides/common/assembly_using-foreman-webhooks.adoc:6]: A webhook is a way for a web page or web application to provide other applications with information. The information is provided immediately. Specific events trigger customized callbacks, such as sending an email when a user registers with a website, or acknowledging online purchases. Because webhooks use HTTP, no new infrastructure need be added to existing web services.

Just a suggestion.

File: guides/common/modules/ref_webhooks-templates.adoc [hunk 6]
```diff
@@ -0,0 +1,6 @@
+[id="webhooks-templates_{context}"]
+= Templates
+
+Webhook HTTP payload must be created via {Project} template syntax. The subject of the event is available via `@object` variable. {Project} ships with an example template for every single event available. For more information, see {ManagingHostsDocURL}appe-Red_Hat_Satellite-Managing_Hosts-Template_Writing_Reference[Template Writing Reference] in _Managing Hosts_ and available template macros and methods, visit `/templates_doc` on a {ProjectServer}.
```
Comment (ianf77) [inline guides/common/modules/ref_webhooks-templates.adoc:4]: Is the link to Managing Hosts here correct? It looks very strange.

Comment (melcorr) [reply guides/common/modules/ref_webhooks-templates.adoc:4]: Please build locally and test @ianf77 
It is best practice.
